src/interfaces/IDelegation.sol

```diff
@@ -13,6 +25,5 @@ interface IDelegation {
 
     function delegate(address delegatee) external;
 
-    function delegateWithSig(address delegatee, uint256 nonce, uint256 expiry, uint8 v, bytes32 r, bytes32 s)
-        external;
+    function delegateWithSig(Delegation memory delegation, Signature calldata signature) external;
```

Comment:
What's the reasoning for putting one as calldata and the other as memory ? IIRC, the memory data location was only useful as a workaround of a solidity bug ?

Reply:
A good question, I reckon they should be both `calldata` right?

Reply:
I remember something like "it's not more expensive because it's put in memory anyway for the keccak" but I tried and it's not true so I change.

Reply:
Unresolving because the signature in the `IDelegation.sol` file is still `memory`